projects/cloud/app/services/project_fetch_service.rb

```diff
@@ -1,16 +1,23 @@
 # frozen_string_literal: true
 
 class ProjectFetchService < ApplicationService
-  attr_reader :name, :account_name
+  module Error
+    Unauthorized = Class.new(StandardError)
+  end
+
+  attr_reader :name, :account_name, :user
 
-  def initialize(name:, account_name:)
+  def initialize(name:, account_name:, user:)
     super()
     @name = name
     @account_name = account_name
+    @user = user
   end
 
   def call
     account = Account.find_by(name: account_name)
-    Project.find_by(account_id: account.id, name: name)
+    project = Project.find_by(account_id: account.id, name: name)
```

Comment:
This might return `nil` and you'll be passing that down to `ProjectPolicy`. I'd instead use `find`, rescue from [this error](https://api.rubyonrails.org/v6.1.4/classes/ActiveRecord/RecordNotFound.html), and return a `Error::ProjectNotFound` error.